waspc/src/Wasp/Analyzer/TypeChecker/TypeError.hs

```diff
@@ -82,22 +82,35 @@ data TypeCoercionErrorReason e
     ReasonDictWrongKeyType String e
   deriving (Eq, Show)
 
-getTypeCoercionErrorMessageAndCtx :: (Type -> TypedExpr -> String) -> TypeCoercionError -> (String, Ctx)
-getTypeCoercionErrorMessageAndCtx getUncoercableTypesMsg (TypeCoercionError (WithCtx ctx texpr) t reason) =
+second :: (b -> d) -> (a, b, c) -> (a, d, c)
+second f (x, y, z) = (x, f y, z)
+
+extractTypeErrorMessagesAndCtx :: (Type -> TypedExpr -> String) -> TypeCoercionError -> (String, [String], Ctx)
```

Comment:
Couldn't come up with a better name for this helper function. Perhaps I'll end up removing it after the comments.